src/controllers/user.js

```diff
@@ -83,4 +84,50 @@ const logout = (req, res) => {
   }
 };
 
-module.exports = { login, register, logout };
+const getUserProfile = async (req, res, next) => {
+  const { username } = req.params;
+  try {
+    const user = await User.findOne({ username }).populate({
+      path: 'donations',
+      populate: {
+        path: 'projectID',
+        model: 'projects',
+      },
+    });
+
+    if (!user) return res.status(404).json({ message: 'User not found' });
+
+    const info = {
+      username: user.username,
+      email: user.email,
+      name: user.name,
+      role: user.role,
+      type: user.type,
+    };
+    const donations = [];
+
+    user.donations.forEach((donation) => {
+      const single = {
+        projectName: donation.projectID.title,
+        donationAmount: donation.amount,
+        donationDate: donation.timestamp,
+      };
+      donations.push(single);
+    });
+
+    await Promise.all(donations);
```

Comment:
since we already have them fetched in the populate no need to await Promise.all() here